### PR TITLE
Release v0.7.0 to main

### PR DESCRIPTION
**Pull request by ed-xmos**
* CHANGED: Tools version from 15.1.4 to 15.2.1
  * CHANGED: Example builds and docs use Ninja instead of nmake under Windows

This will require a change to sw_avona_main view file afterwards and also a merge back to develop.

### CODE REVIEW

File: doc/README.rst [hunk 1]
```diff
@@ -12,13 +12,7 @@ Building Documentation
...
 =============
 
-Install `Docker <https://www.docker.com/>`_.
-
-Pull the docker container:
-
-.. code-block:: console
-
-    $ docker pull ghcr.io/xmos/doc_builder:main
+Use the `xmosdoc tool <https://github.com/xmos/xmosdoc>`_ either via docker or install it into a pip environment.
```
Comment (lucianomartin) [inline doc/README.rst:15]: We need to add some information to explain xmosdoc is for XMOS users only. We can add something like we added at the beginning of the readme in fwk_core:

https://github.com/xmos/fwk_core/blob/develop/doc/README.rst

**Review by lucianomartin — COMMENTED** [review]
Just one comment.

File: doc/getting_started/index.rst [hunk 2]
```diff
@@ -42,9 +44,8 @@ procedure is currently supported on MacOS, Linux and Windows.
 
       .. code-block:: console
 
-
         # make sure you have the patch command available
-        cmake -G "NMake Makefiles" -S.. -DCMAKE_TOOLCHAIN_FILE=../xmos_cmake_toolchain/xs3a.cmake
+        cmake -G "Ninja" --toolchain  ../xmos_cmake_toolchain/xs3a.cmake -S..
```
Comment (mbanth) [inline doc/getting_started/index.rst:48]: In other documentation, we have not included the `-S` option explicitly.  From the [cmake documentation](https://cmake.org/cmake/help/latest/manual/cmake.1.html), it looks like a path without an explicit preceeding `-S` behaves the same as one with this option.  I prefer that we settle on a consistent use of this option, although I do not have a strong opinion either way about which is preferred.

This comment applies to multiple files in this pull request.

File: doc/install_ninja_rst.inc [hunk 3]
```diff
@@ -0,0 +1,19 @@
+It is highly recommended to use `Ninja` as the make system under
+Windows. Not only has `Ninja` observed to be faster than MSVC `nmake`, it also
```
Comment (mbanth) [inline doc/install_ninja_rst.inc:2]: The phrase `Not only has Ninja observed to be faster ...` doesn't make sense.  Perhaps `Not only has Ninja been observed to be faster ...` is what is meant.

File: doc/install_ninja_rst.inc [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+works around an XMOS tools bug where certain path names may cause an issue with
+the XMOS compiler under windows.  This firmware has been tested with `Ninja` version v1.11.1.
```
Comment (mbanth) [inline doc/install_ninja_rst.inc:4]: Please provide an initial captial on `Windows`.

File: doc/install_ninja_rst.inc [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+To install `Ninja`, follow these steps:
+
+-  Download ``ninja.exe`` from https://github.com/ninja-build/ninja/releases and save
```
Comment (mbanth) [inline doc/install_ninja_rst.inc:8]: Please make the URL an actionable link.

File: doc/install_ninja_rst.inc [hunk 6]
```diff
@@ -0,0 +1,19 @@
+It is highly recommended to use `Ninja` as the make system under
+Windows. Not only has `Ninja` observed to be faster than MSVC `nmake`, it also
+works around an XMOS tools bug where certain path names may cause an issue with
+the XMOS compiler under windows.  This firmware has been tested with `Ninja` version v1.11.1.
+
+To install `Ninja`, follow these steps:
+
+-  Download ``ninja.exe`` from https://github.com/ninja-build/ninja/releases and save 
+   to a suitable location such as ``C:\Program Files\ninja``.
+-  Ensure `Ninja` is on the command line path. It may be added to the systempath using the following command:
```
Comment (mbanth) [inline doc/install_ninja_rst.inc:10]: `systempath` should be `system path`.

**Review by mbanth — CHANGES_REQUESTED** [review]

File: doc/getting_started/index.rst (hunk 2 above)
Comment (ed-xmos) [reply doc/getting_started/index.rst:48]: Yes I have never used the -S before - but I tested it and it worked fine so kept it in line with the rest of the documentation.

File: doc/install_ninja_rst.inc (hunk 3 above)
Comment (ed-xmos) [reply doc/install_ninja_rst.inc:2]: It took me 3 reads to see the missing 'been'. I have now added it! Must be one of those optical illusions.

File: doc/install_ninja_rst.inc (hunk 5 above)
Comment (ed-xmos) [reply doc/install_ninja_rst.inc:8]: Done

File: doc/install_ninja_rst.inc (hunk 6 above)
Comment (ed-xmos) [reply doc/install_ninja_rst.inc:10]: good spot

File: doc/README.rst (hunk 1 above)
Comment (ed-xmos) [reply doc/README.rst:15]: Well reminded. Fixed

**Review by mbanth — APPROVED** [review]
Looks good now.

**Pull request merged**